Promise/Promise.swift

```diff
@@ -9,74 +9,89 @@
 import Foundation
 
 struct Callback<Value> {
-    let onFulfilled: (Value -> ())
-    let onRejected: (ErrorType -> ())
-    let queue: dispatch_queue_t
+    let onFulfilled: (Value) -> Void
+    let onRejected: (Error) -> Void
+    let queue: DispatchQueue
```

Comment:
if it's possible, i'd like to keep these functions as `(Value) -> ()`. sorry again that there's no style guide.